common/beerocks/bwl/include/bwl/base_wlan_hal.h

```diff
@@ -92,6 +92,15 @@ class base_wlan_hal {
      */
     virtual bool process_ext_events() = 0;
 
+    /*!
+     * Process incoming nl events from the underlying hardware/middleware.
+     * This method should be called if the file descriptor returned by
+     * get_nl_events_fd() generated an event.
+     *
+     * @return true on success or false on error.
+     */
+    virtual bool process_nl_events() = 0;
```

Comment:
If you add such a pure virtual function, you need to also add an implementation to each subclass (nl80211 and dummy in addition to dwpal).

However, it doesn't seem to be called anywhere...

Reply:
its an INFRA PR.
It will be called from monitor_thread.cpp in the next PR.